### DISCUSSION TITLE
[keycloak quarkus 18.0.0] running against db: empty page

### DISCUSSION THREAD

**Discussion by rbaumgar** [keycloak/keycloak#12402] [Archive]
I try to run against a db (mariadb) in OpenShift 4.10 / Kubernetes 1.23
start page works.
when click "Administration Console" (/admin/master/console/) I get an empty page.

Inspect shows following
step1.html (pending)
![image](https://user-images.githubusercontent.com/19689547/172630482-23a0c4c9-f880-412a-8d90-5d90e376deaa.png)

this is the pod log
```
Changes detected in configuration. Updating the server image.
Updating the configuration and installing your custom providers, if any. Please wait.
Server configuration updated and persisted. Run the following command to review the configuration:
kc.sh show-config
Next time you run the server, just run:
kc.sh start --db-url-host=mariadb --db-url-database=db --db-username=dbuser --db-password=******* --hostname=rhsso-demo.apps.ocp4.openshift.freeddns.org --http-enabled=true --log-level=warn
2022-06-08 13:30:21,590 WARN [org.infinispan.CONFIG] (keycloak-cache-init) ISPN000569: Unable to persist Infinispan internal caches as no global state enabled
2022-06-08 13:30:21,611 WARN [org.infinispan.PERSISTENCE] (keycloak-cache-init) ISPN000554: jboss-marshalling is deprecated and planned for removal
2022-06-08 13:30:22,263 WARN [org.jgroups.protocols.UDP] (keycloak-cache-init) JGRP000015: the send buffer of socket MulticastSocket was set to 1.00MB, but the OS only allocated 212.99KB
2022-06-08 13:30:22,263 WARN [org.jgroups.protocols.UDP] (keycloak-cache-init) JGRP000015: the receive buffer of socket MulticastSocket was set to 20.00MB, but the OS only allocated 212.99KB
2022-06-08 13:30:22,264 WARN [org.jgroups.protocols.UDP] (keycloak-cache-init) JGRP000015: the send buffer of socket MulticastSocket was set to 1.00MB, but the OS only allocated 212.99KB
2022-06-08 13:30:22,264 WARN [org.jgroups.protocols.UDP] (keycloak-cache-init) JGRP000015: the receive buffer of socket MulticastSocket was set to 25.00MB, but the OS only allocated 212.99KB
```

this is my pod config
```
spec:
  restartPolicy: Always
  containers:
    - resources: {}
      terminationMessagePath: /dev/termination-log
      name: keycloak
      env:
        - name: KEYCLOAK_ADMIN
          valueFrom:
            secretKeyRef:
              name: keycloak-secret
              key: keycloak-user
        - name: KEYCLOAK_ADMIN_PASSWORD
          valueFrom:
            secretKeyRef:
              name: keycloak-secret
              key: keycloak-password
      ports:
        - containerPort: 8080
          protocol: TCP
      image: 'quay.io/keycloak/keycloak:18.0.0'
      args:
        - >-
          start --auto-build --db mariadb --db-url-host mariadb
          --db-url-database db --db-username dbuser --db-password dbpassword
          --hostname rhsso-demo.apps.ocp4.openshift.freeddns.org --http-enabled
          true --log-level warn --features token-exchange
```

**Comment by DGuhr** [comment]
Hi @rbaumgar when you want to access keycloak in prod mode with http, you have to also set `--hostname-strict-https=false`. We missed to document that, but it's in upstream (ref: https://github.com/keycloak/keycloak/blob/main/docs/guides/src/main/server/hostname.adoc#accessing-keycloak-in-production-mode-using-http ) and will make it into the next docs release. Hope this helps :)

**Comment by rbaumgar** [comment]
@DGuhr 
sorry does not really help.
now I got following error
`Refused to frame 'http://rhsso-demo.apps.ocp4.openshift.freeddns.org/' because it violates the following Content Security Policy directive: "frame-src 'self'".`

I have already front enden the url with a https!
`oc create route edge --service rhsso`

When I remove "--http-enabled true", I get a startup problem because I do not present a certificate. This is not required, becaue I am using edge terminated ssl.

**Reply by DGuhr** [reply]
then the aforementioned strict-https together with `--proxy=edge` should do the trick imo

**Comment by rbaumgar** [comment]
@DGuhr 

You are right! Works! I did not saw this parameter.
I believe should be documented, somehow.

**Reply by DGuhr** [reply]
https://www.keycloak.org/server/reverseproxy ;) Happy it works for you :)

**Comment by rbaumgar** [comment]
@DGuhr 
you are correct.
I meant the combination...
When you setup production with edge terminated you need:
```
--hostname <route-host-name>
--hostname-strict-https=false 
--proxy edge 
--http-enabled true
```
I believe this will be the default more less.
